### PR TITLE
CloudWatch: Add field level comments to schema

### LINKED ISSUE #66062: CloudWatch: Add field level comments in DataQuery schema
Add field level comments in DataQuery schema.

### PR DESCRIPTION
**Pull request by kevinwcyu**
**What is this feature?**

Adds field level comments to the CloudWatch DataQuery schema

**Which issue(s) does this PR fix?**:

Fixes #66062 

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/plugins/datasource/cloudwatch/dataquery.gen.ts [hunk 1]
```diff
@@ -31,23 +55,50 @@ export const defaultMetricStat: Partial<MetricStat> = {
...
+ * A name/value pair that is part of the identity of a metric. For example, you can get statistics for a specific EC2 instance by specifying the InstanceId dimension when you search for metrics.
+ */
 export type Dimensions = Record<string, (string | Array<string>)>;
 
+/**
+ * Shape of a CloudWatch Metrics query
+ */
 export interface CloudWatchMetricsQuery extends common.DataQuery, MetricStat {
+  /**
+   * To be deprecated. Use alias
```
Comment (sarahzinger) [inline public/app/plugins/datasource/cloudwatch/dataquery.gen.ts:68]: use label?

**Review by sarahzinger — COMMENTED** [review]
This is really awesome stuff!

File: public/app/plugins/datasource/cloudwatch/dataquery.gen.ts (hunk 1 above)
Comment (kevinwcyu) [reply public/app/plugins/datasource/cloudwatch/dataquery.gen.ts:68]: haha, yea

File: public/app/plugins/datasource/cloudwatch/dataquery.gen.ts [hunk 2]
```diff
@@ -168,16 +249,43 @@ export const defaultCloudWatchLogsQuery: Partial<CloudWatchLogsQuery> = {
...
+  /**
+   * Use this parameter to filter the results of the operation to only those alarms that use a certain alarm action prefix.
+   */
   actionPrefix?: string;
+  /**
+   * Use this parameter to filter the results of the operation to only those alarms that use a certain alarm name prefix.
+   */
   alarmNamePrefix?: string;
+  /**
+   * Enable matching on the prefix of the action name or alarm name, specify the prefixes with actionPrefix and/or alarmNamePrefix
```
[the remark this answers is not in the document]
Comment (kevinwcyu) [reply public/app/plugins/datasource/cloudwatch/dataquery.gen.ts:283]: this wouldn't be used, i'll update the comment to clarify

File: public/app/plugins/datasource/cloudwatch/dataquery.gen.ts [hunk 3]
```diff
@@ -168,16 +249,43 @@ export const defaultCloudWatchLogsQuery: Partial<CloudWatchLogsQuery> = {
...
+   */
   name: string;
 }
 
+/**
+ * Shape of a CloudWatch Annotation query
+ */
 export interface CloudWatchAnnotationQuery extends common.DataQuery, MetricStat {
+  /**
+   * Use this parameter to filter the results of the operation to only those alarms that use a certain alarm action prefix.
```
[the remark this answers is not in the document]
Comment (kevinwcyu) [reply public/app/plugins/datasource/cloudwatch/dataquery.gen.ts:275]: yea, i could put an example in.

File: public/app/plugins/datasource/cloudwatch/dataquery.gen.ts [hunk 4]
```diff
@@ -109,17 +181,11 @@ export interface QueryEditorGroupByExpression {
 }
 
 export interface QueryEditorOperatorExpression {
-  /**
-   * TS type is operator: QueryEditorOperator<QueryEditorOperatorValueType>, extended in veneer
-   */
```
[the remark this answers is not in the document]
Comment (kevinwcyu) [reply public/app/plugins/datasource/cloudwatch/dataquery.gen.ts:114]: i deleted it in the .cue file and this is autogenerated so it ended up being removed. although i don't think i should've removed it.

a veneer is a handwritten type on top of the generated code. in this file we import the generated `QueryEditorOperator` and then extend it. this is done when we can't fully express the type in the schema just yet.

https://github.com/grafana/grafana/blob/a2b97547a6fec749791d02a44de781a29205b6fa/public/app/plugins/datasource/cloudwatch/expressions.ts#L18

here's an example of when a veneer can be removed and shows an example of something that previously couldn't be expressed in the schema that we needed a veneer for https://github.com/grafana/grafana/pull/66950/files

File: public/app/plugins/datasource/cloudwatch/dataquery.gen.ts [hunk 5]
```diff
@@ -31,23 +55,50 @@ export const defaultMetricStat: Partial<MetricStat> = {
...
+  /**
+   * Whether a query is a Metrics, Logs, or Annotations query
+   */
   queryMode?: CloudWatchQueryMode;
+  /**
+   * When the metric query type is `metricQueryType` is set to `Query` and the `metricEditorMode` is set to `Builder`, this field is used to build up an object representation of a SQL query.
+   */
   sql?: SQLExpression;
+  /**
+   * When the metric query type is `metricQueryType` is set to `Query`, this field is used to specify the query string.
```
[the remark this answers is not in the document]
Comment (kevinwcyu) [reply public/app/plugins/datasource/cloudwatch/dataquery.gen.ts:100]: it's used in both editor modes. in builder mode it generates a sql string from the sql object and then it's assigned to `sqlExpression`

https://github.com/grafana/grafana/blob/a2b97547a6fec749791d02a44de781a29205b6fa/public/app/plugins/datasource/cloudwatch/components/SQLBuilderEditor/SQLBuilderEditor.tsx#L28-L31

**Review by sarahzinger — APPROVED** [review]
Seems cool to me! Guessing this would be a 10.0.x milestone then? Not sure if we're adding these things to changelog or not.

**Review by idastambuk — APPROVED** [review]

**Pull request merged**